templates/web/base/tokens/confirm_problem.html

```diff
@@ -2,7 +2,13 @@
 
 <div class="confirmation-header">
 
-  <h1><a href="[% c.cobrand.relative_url_for_report( report ) %][% report.url %]">[% report.title %]</a></h1>
+  <h1><a href="[% c.cobrand.relative_url_for_report( report ) %][% report.url %]">
+    [% IF report.non_public AND NOT can_view_report %]
```

Comment:
`[%~` so as to not have whitespace at the start of the link? (And similar throughout/elsewhere)